sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/UuidDeduplicationTransform.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.gcp.pubsublite;
+
+import com.google.cloud.pubsublite.SequencedMessage;
+import java.math.BigInteger;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.ProcessFunction;
+import org.apache.beam.sdk.transforms.Values;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TypeDescriptor;
+
+class UuidDeduplicationTransform
+    extends PTransform<PCollection<SequencedMessage>, PCollection<SequencedMessage>> {
+  private final UuidDeduplicationOptions options;
+
+  UuidDeduplicationTransform(UuidDeduplicationOptions options) {
+    this.options = options;
+  }
+
+  @Override
+  public PCollection<SequencedMessage> expand(PCollection<SequencedMessage> input) {
+    input.getPipeline().getCoderRegistry().registerCoderForClass(Uuid.class, Uuid.getCoder());
```

Comment:
Registering a coder as part of expand is weird, and will be confusing if the user uses these types elsewhere in the pipeline.  You can instead use the @DefaultCoder annotation on Uuid and SequencedMessage to set their default coders globally.

Reply:
Uuid and SequencedMessage are part of the Pub/Sub Lite client library, not beam. Is there a way to register a default coder somewhere not in the type definition?

Reply:
ah sorry, thatts only true of SequencedMessage. Will make this change for Uuid.

Reply:
done for uuid, is there a way to set this globally on the SequencedMessage type that comes from the Pub/Sub Lite java client library?

Reply:
Yeah, you can create a CoderProviderRegistrar.  There's an example from another IO solving the same problem here: https://github.com/apache/beam/blob/master/sdks/java/io/hbase/src/main/java/org/apache/beam/sdk/io/hbase/HBaseCoderProviderRegistrar.java
In that provider, add default coders for SequencedMessage and Message

Also, remove the registerCoderForClass calls here

Reply:
Done.